src/gui/socketapi.cpp

```diff
@@ -514,6 +508,39 @@ void SocketApi::command_SHARE_MENU_TITLE(const QString &, SocketListener *listen
     listener->sendMessage(QLatin1String("SHARE_MENU_TITLE:") + tr("Share with %1", "parameter is ownCloud").arg(Theme::instance()->appNameGUI()));
 }
 
+void SocketApi::command_COPY_PRIVATE_LINK(const QString &localFile, SocketListener *)
+{
+    auto url = getPrivateLinkUrl(localFile);
+    if (!url.isEmpty()) {
+        QApplication::clipboard()->setText(url.toString());
+    }
+}
+
+void SocketApi::command_EMAIL_PRIVATE_LINK(const QString &localFile, SocketListener *)
+{
+    auto url = getPrivateLinkUrl(localFile);
+    if (!url.isEmpty()) {
+        Utility::openEmailComposer(
+            tr("I shared something with you"),
+            url.toString(),
+            0);
+    }
+}
+
+void SocketApi::command_GET_STRINGS(const QString &, SocketListener *listener)
+{
+    static auto map = std::initializer_list<std::pair<const char *, QString>>{
+        { "SHARE_MENU_TITLE", tr("Share with %1...", "parameter is ownCloud").arg(Theme::instance()->appNameGUI()) },
```

Comment:
Putting the appName in both the submenu title and the share entry looks a bit redundant now:
![screen shot 2017-07-04 at 13 37 16](https://user-images.githubusercontent.com/839935/27839252-f61d3a28-60bd-11e7-898e-4b05f5f3fa4c.png)

Maybe we could name it just "Share..."?